### PR TITLE
US585006: A new API "batchStatus" to check status of the batch

### PR DESCRIPTION
**Pull request by raghavendran-gopalakrishnan**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=585006

### CODE REVIEW

File: staging-service-acceptance-tests/src/test/java/com/github/cafdataprocessing/services/staging/StagingServiceIT.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 {
 
     private static final String STAGING_SERVICE_URI = System.getenv("staging-service");
-//        private static final String STAGING_SERVICE_URI = "http://localhost:8080";
+//  private static final String STAGING_SERVICE_URI = "http://localhost:8080";
```
Comment (kusumaghoshdastidar) [inline staging-service-acceptance-tests/src/test/java/com/github/cafdataprocessing/services/staging/StagingServiceIT.java]: Remove unrelated change

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java [hunk 2]
```diff
@@ -317,7 +328,7 @@ private Optional<Path> getTenantInprogressDirectorySafely(final String tenantIdF
     /**
      * Returns a list of the path to all sub files in all folders provide
      *
-     * @param tenantFolders List of Path objects that represent all current tenant in_progress folders
+     * @param tenantFolder List of Path objects that represent all current tenant in_progress folders
```
Comment (kusumaghoshdastidar) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java]: Remove unrelated change

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java [hunk 3]
```diff
@@ -282,6 +282,17 @@ public void cleanUpStaleInprogressBatches()
         }
     }
 
+    @Override
+    public String getBatchStatus(TenantId tenantId, BatchId batchId) throws BatchNotFoundException {
```
Comment (kusumaghoshdastidar) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java]: Make params final

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java [hunk 4]
```diff
@@ -282,6 +282,17 @@ public void cleanUpStaleInprogressBatches()
         }
     }
 
+    @Override
```
Comment (kusumaghoshdastidar) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java:286]: Follow code formatting to match the rest of the file

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java [hunk 5]
```diff
@@ -132,6 +132,29 @@ public ResponseEntity<Void> deleteBatch(
...
 
+    @Override
+    public ResponseEntity<StatusResponse> getBatchStatus(
+            @ApiParam(value = "Identifies the tenant making the request.", required = true)
+            @RequestHeader(value = "X-TENANT-ID", required = true) String X_TENANT_ID,
+            @ApiParam(value = "Identifies the batch.", required = true)
+            @RequestParam("batchId") String batchId) {
+        final StatusResponse status = new StatusResponse();
+        try {
+            String statusResponse = batchDao.getBatchStatus(new TenantId(X_TENANT_ID), new BatchId(batchId));
```
Comment (kusumaghoshdastidar) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]:
```suggestion
            final String statusResponse = batchDao.getBatchStatus(new TenantId(X_TENANT_ID), new BatchId(batchId));
```

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java [hunk 6]
```diff
@@ -132,6 +132,29 @@ public ResponseEntity<Void> deleteBatch(
...
+            status.setMessage(statusResponse);
+            return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(status);
+        } catch (InvalidTenantIdException | InvalidBatchIdException ex){
```
Comment (kusumaghoshdastidar) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: Separate exceptions so the response error messages can be specific
Make final

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java [hunk 7]
```diff
@@ -132,6 +132,29 @@ public ResponseEntity<Void> deleteBatch(
...
+            LOGGER.error("Invalid X-TENANT-ID or Invalid Batch Id", ex);
+            throw new WebMvcHandledRuntimeException(HttpStatus.BAD_REQUEST, ex.getMessage());
+        } catch (final BatchNotFoundException ex) {
+            LOGGER.error("Batch not found.", ex);
+            throw new WebMvcHandledRuntimeException(HttpStatus.NOT_FOUND, ex.getMessage());
+        } catch (StagingException ex) {
```
Comment (kusumaghoshdastidar) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]:
```suggestion
        } catch (final StagingException ex) {
```

File: staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchPathProvider.java [hunk 8]
```diff
@@ -109,6 +110,11 @@ public Path getTenantInprogressDirectory(final TenantId tenantId)
         return getPathForTenant(tenantId).resolve(INPROGRESS_FOLDER);
     }
 
+    public boolean isBatchInProgress(final TenantId tenantId, final BatchId batchId){
```
Comment (kusumaghoshdastidar) [inline staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchPathProvider.java]: Follow code formatting to match the rest of the file

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/BatchDao.java [hunk 9]
```diff
@@ -40,4 +40,6 @@ List<String> getBatches(TenantId tenantId, @Size(min = 1, max = 256) @Valid Stri
 
     void cleanUpStaleInprogressBatches();
 
+    String getBatchStatus(TenantId tenantId, @Size(max = 1) BatchId batchId) throws BatchNotFoundException, StagingException;
```
Comment (kusumaghoshdastidar) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/BatchDao.java]: Could the status be an enum?

Comment (raghavgopal) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/BatchDao.java]: Ya it could be, thought of returning the status with the BatchId appended., if that's not required I can return enum or final constant.

Comment (raghavgopal) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/BatchDao.java]: Addressed.

**Comment by kusumaghoshdastidar** [comment]
There might be abandoned/stale batches in the "in-progress" folder which would be cleaned up periodically if **CAF_STAGING_SERVICE_SKIP_FILE_CLEANUP** is set to `false`.
But if **CAF_STAGING_SERVICE_SKIP_FILE_CLEANUP** is set to `true` there may be batch folders/subfolders that are present past the **CAF_STAGING_SERVICE_FILE_AGE_THRESHOLD** time. These could be assumed to be `stale` or `abandoned`.
Refer to this ticket for details: https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=359243

Include an `ABANDONED` status for the scenario with `CAF_STAGING_SERVICE_SKIP_FILE_CLEANUP=true`

**Review by kusumaghoshdastidar — CHANGES_REQUESTED** [review]
There might be abandoned/stale batches in the "in-progress" folder which would be cleaned up periodically if **CAF_STAGING_SERVICE_SKIP_FILE_CLEANUP** is set to `false`.
But if **CAF_STAGING_SERVICE_SKIP_FILE_CLEANUP** is set to `true` there may be batch folders/subfolders that are present past the **CAF_STAGING_SERVICE_FILE_AGE_THRESHOLD** time. These could be assumed to be `stale` or `abandoned`.
Refer to this ticket for details: https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=359243

Include an `ABANDONED` status for the scenario with `CAF_STAGING_SERVICE_SKIP_FILE_CLEANUP=true`

**Review by kusumaghoshdastidar — APPROVED** [review]

**Comment by dermot-hardy** [comment]
I've just looked at this and I see that 2 60.0 MB files have been checked in.  This is a Git repository.  Surely this triggered warnings.  Please look up and understand the consequences of checking large files into Git repositories.  Git LFS is not enabled in this repository.  Not only do I have to back this out I'm going to have to do a force push to remove it from the history.

@kusumaghoshdastidar This should not have made it through code review.
